course/Program.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Newtonsoft.Json;
+
+namespace Course
+{
+    public enum Side
+    {
+        North,
+        South,
+        West,
+        East,
+    }
+
+    class Program
+    {
+        static void Main()
+        {
+            var persons = JsonConvert.DeserializeObject<IEnumerable<Person>>(File.ReadAllText("data.json"));
+
+            FarthestPerson(persons, Side.West);
+            FarthestPerson(persons);
+            Console.WriteLine("\n");
+            SimilarAboutPeople(persons);
+            Console.WriteLine("\n");
+            SameFriendPeople(persons);
+            Console.WriteLine("\n");
+            CompanyWithMostPeople(persons);
+        }
+
+        static void FarthestPerson(IEnumerable<Person> persons, Side side = Side.North)
+        {
+            var person = side switch
+            {
+                Side.South => persons.MinBy(x => x.Latitude),
+                Side.East => persons.MaxBy(x => x.Longitude),
+                Side.West => persons.MinBy(x => x.Longitude),
+                _ => persons.MaxBy(x => x.Latitude),
```

Comment:
in case if I pass as Side, e.g. (Side)42 it will search for the northeast person, but should be exception

Reply:
right, thanks